### PR TITLE
Automized setup (pip) to drag-in build time dependencies (numpy, cython)

### LINKED ISSUE #933: Automatic dependency resolvation with pip
I wonder if you would accept a PR which drags in dependencies like NumPy and Cython during installation with pip. If one installs a mdtraj dependent project in a clean environment the order of installation might get resolved wrong, if these are not declared in a setuptools specific manner.

This would not interfere with conda, since automatic downloading has been blocked in recent versions of conda-build anyway.

If one opts in cython during setup time, one should considerate to request the users to have at least setuptools >= 18, because since this version the hassle of cython builds has been fixed.

What do you think?

### PR DESCRIPTION
**Pull request by marscher**
* NumPy includes and Cython extensions are handled transparently.
* Requires setuptools >= 18

closes #933

### CODE REVIEW

**Comment by rmcgibbo** [comment]
Unfortunately, inclusion the numpy include directories (including `np.get_include()` in the `Extension` configuration) is not optional -- note the failures on Travis. The cython extensions use typedefs from `npy_common.h` and such, and need to be compiled against numpy.

**Comment by marscher** [comment]
Added numpy inc dir, but now Appveyor Py3 fails. I can not see an useful error message.

**Comment by swails** [comment]
> Added numpy inc dir, but now Appveyor Py3 fails. I can not see an useful error message.

I believe it's *been* failing.  This is current master: https://ci.appveyor.com/project/rmcgibbo/mdtraj/branch/master

**Comment by rmcgibbo** [comment]
What is the thing that was fixed / added in setuptools 18 that makes this work?

**Comment by marscher** [comment]
The transparent Cython handling is new in this version. I guess they're
searching the sources list for pyx files and cythonize them on the fly.
This way setuptools is able to install cython during setup time
(setup_requires dependency).

https://bitbucket.org/pypa/setuptools/issue/288

File: setup.py [hunk 1]
```diff
@@ -172,13 +147,13 @@ def rmsd_extensions():
                            'mdtraj/rmsd/_lprmsd.pyx'],
                        language='c++',
                        include_dirs=[
-                           'mdtraj/rmsd/include', numpy.get_include()],
+                           'mdtraj/rmsd/include',
+                                    ],
```
Comment (mpharrigan) [inline setup.py]: this formatting looks weird

**Comment by rmcgibbo** [comment]
Sweet.

**Pull request merged**